### ISSUE TITLE
QueryBuilderMaterial is throwing "Element type is invalid: expected a string or a class/function but got: object"

### ISSUE THREAD

**Issue by zolbayars** [react-querybuilder/react-querybuilder#601]
Hi all, 
`<QueryBuilder>` is working fine but as soon as I wrap it with `<QueryBuilderMaterial>` it throws: 

```
Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: object.
    at MaterialValueSelector (webpack-internal:///(ssr)/./node_modules/@react-querybuilder/material/dist/react-querybuilder_material.mjs:183:32)
    at RuleGroupHeaderComponents (webpack-internal:///(ssr)/./node_modules/react-querybuilder/dist/react-querybuilder.mjs:16514:410)
    at div
    at div
    at RuleGroup (webpack-internal:///(ssr)/./node_modules/react-querybuilder/dist/react-querybuilder.mjs:16478:12)
    at div
    at QueryBuilder (webpack-internal:///(ssr)/./node_modules/react-querybuilder/dist/react-querybuilder.mjs:16285:12)
    at eval (webpack-internal:///(ssr)/./node_modules/react-querybuilder/dist/react-querybuilder.mjs:1654:77)
    at QueryBuilderMaterial (webpack-internal:///(ssr)/./node_modules/@react-querybuilder/material/dist/react-querybuilder_material.mjs:439:46)
    at main
    at Home (webpack-internal:///(ssr)/./app/page.tsx:41:78)
    at StaticGenerationSearchParamsBailoutProvider (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/static-generation-searchparams-bailout-provider.js:15:11)
    at Lazy
    at InnerLayoutRouter (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/layout-router.js:239:11)
    at RedirectErrorBoundary (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/redirect-boundary.js:71:9)
    at RedirectBoundary (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/redirect-boundary.js:79:11)
    at NotFoundErrorBoundary (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/not-found-boundary.js:54:9)
    at NotFoundBoundary (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/not-found-boundary.js:62:11)
    at LoadingBoundary (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/layout-router.js:343:11)
    at ErrorBoundary (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/error-boundary.js:128:11)
    at InnerScrollAndFocusHandler (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/layout-router.js:151:9)
    at ScrollAndFocusHandler (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/layout-router.js:226:11)
    at RenderFromTemplateContext (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/render-from-template-context.js:15:44)
    at Lazy
    at OuterLayoutRouter (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/layout-router.js:352:11)
    at Lazy
    at body
    at html
    at RedirectErrorBoundary (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/redirect-boundary.js:71:9)
    at RedirectBoundary (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/redirect-boundary.js:79:11)
    at ReactDevOverlay (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/react-dev-overlay/internal/ReactDevOverlay.js:66:9)
    at HotReload (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/react-dev-overlay/hot-reloader-client.js:296:11)
    at Router (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/app-router.js:165:11)
    at ErrorBoundaryHandler (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/error-boundary.js:99:9)
    at ErrorBoundary (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/error-boundary.js:128:11)
    at AppRouter (webpack-internal:///(ssr)/./node_modules/next/dist/client/components/app-router.js:436:13)
    at Lazy
    at Lazy
    at ...\tmp\test-querybuilder\node_modules\next\dist\compiled\next-server\app-page.runtime.dev.js:35:380095
    at ...\tmp\test-querybuilder\node_modules\next\dist\compiled\next-server\app-page.runtime.dev.js:35:380095
    at ServerInsertedHTMLProvider (...\tmp\test-querybuilder\node_modules\next\dist\compiled\next-server\app-page.runtime.dev.js:38:17348)
```

### Environment/deps

```
    "@react-querybuilder/material": "^6.5.4",
    "next": "14.0.3",
    "react": "^18",
    "react-dom": "^18",
    "react-querybuilder": "^6.5.4"
```

### Steps to reproduce

1. `npx create-next-app@latest` to create a new Nextjs project
2. `npm i react-querybuilder @react-querybuilder/material` to install the packages
3. Replace the `app/page.tsx` with the sample code: 

```
"use client";

import { useState } from "react";
import type { Field, RuleGroupType } from "react-querybuilder";
import { QueryBuilder } from "react-querybuilder";
import { QueryBuilderMaterial } from "@react-querybuilder/material";

const fields: Field[] = [
  { name: "firstName", label: "First Name" },
  { name: "lastName", label: "Last Name" },
];

const initialQuery: RuleGroupType = {
  combinator: "and",
  rules: [
    { field: "firstName", operator: "beginsWith", value: "Stev" },
    { field: "lastName", operator: "in", value: "Vai,Vaughan" },
  ],
};

export default function Home() {
  const [query, setQuery] = useState(initialQuery);

  return (
    <main className="flex min-h-screen flex-col items-center justify-between p-24">
      <QueryBuilderMaterial>
        <QueryBuilder
          fields={fields}
          query={query}
          onQueryChange={(value) => setQuery(value)}
        />
      </QueryBuilderMaterial>
    </main>
  );
}
```
4. `npm run dev` to start the service 
5. Look at the console log and the server log 

On top of the above issue, it's throwing `Warning: Each child in a list should have a unique "key" prop.` as well.

**Comment by jakeboone02** [comment]
I'm looking into this. My suspicion is it's something to do with RSC but I'll confirm.

**Comment by jakeboone02** [comment]
I followed your instructions (mostly), refactored some stuff (probably unnecessarily) and uploaded [this sample repo](https://github.com/jakeboone02/rqb-next-bug-601). I used RQB `v7.0.0-alpha.2` to make sure things still worked in v7 and also to make sure it wasn't a v6-only bug.

If you look at [app/QB.tsx](https://github.com/jakeboone02/rqb-next-bug-601/blob/087a917d8e315aa65d350c91891d08df37c53b7b/app/QB.tsx#L3-L25), you'll see I had to preload the MUI components in a `"use client"` file. I'm not sure why this is an issue with MUI only and not any other compatibility packages, but it's a problem I can't seem to get around. [Docs mention it here](https://react-querybuilder.js.org/docs/next/compat#preload-mui-components) (I just realized the v7 docs are missing some of the components, but you can see the full list in that sample repo).

That sample repo doesn't throw any server errors for `npm run dev`, but it does still have the `'unique "key" prop'` error in the browser console. It also throws a warning about "Extra attributes from the server" which I'm sure is just a harmless dev-mode thing related to RSC, but if you know a way to avoid that please let me know.

**Comment by zolbayars** [comment]
That worked! Thanks a lot, @jakeboone02 

(Before creating this issue, I had looked into the doc about preloading, but disregarded it due to the tip `As of v6.2.0, MUI components will be loaded synchronously. Preloading the components is no longer necessary and there will be no flash of unstyled content.` since I was using `v6.5.4` 🙂)

Btw, is there a way to get rid of the `Each child in a list should have a unique "key" prop.` and `Extra attributes from the server: data-rule-id` warnings? :)

**Comment by jakeboone02** [comment]
I'll have to update that note about synchronous loading to include an exception for RSC environments.

I'm really not sure why the `unique "key" prop` error persists. As far as traditional (non-RSC) environments go, it was fixed in #585 and released with [v6.5.4](https://github.com/react-querybuilder/react-querybuilder/blob/main/CHANGELOG.md#v654---2023-11-04).

The "extra attributes" thing is RSC-related so I'm unfamiliar with it. I'm going to leave it for now unless you can find something online that mentions how to avoid it.
